app/uk/gov/hmrc/eusubsidycompliance/controllers/UndertakingController.scala

```diff
@@ -49,9 +52,10 @@ class UndertakingController @Inject()(
   def create: Action[JsValue] = Action.async(parse.json) { implicit request =>
     implicit val uF = Json.format[Undertaking]
     withJsonBody[Undertaking] { undertaking: Undertaking =>
-      eis.createUndertaking(undertaking).map{ ref =>
-        Ok(Json.toJson(ref))
-      }
+      for {
+        ref <- eis.createUndertaking(undertaking)
```

Comment:
one quick question may be the dumb one, when an undertaking is created , then a subsidy is also created? or updateSubsidy do insert and update?

Reply:
UpdateSubsidy will insert it if it's not there. We could rename to upsert?

Reply:
sure, if you like I just wanted to confirm that :)